swiftwinrt/Resources/Support/Error.swift

```diff
@@ -104,8 +104,15 @@ public var E_XAMLPARSEFAILED : WinSDK.HRESULT {
 private func getErrorDescription(expecting hr: HRESULT) -> String? {
   var errorInfo: UnsafeMutablePointer<IRestrictedErrorInfo>?
   guard GetRestrictedErrorInfo(&errorInfo) == S_OK else { return nil }
```

Comment:
@stevenbrix , we could also call `RoGetMatchingRestrictedErrorInfo`, but it has slightly different semantics:

> The function checks to see if current error info matches the hrIn value passed in and, if not, it originates a matching error info.

It might be useful still as it guarantees we'll store an `IRestrictedErrorInfo` with the HRESULT, though potentially a generic message. Thoughts?

Reply:
i've never heard of that, but it is interesting. the generic message is all we use today, so that is probably ok.

it could be useful since if the caller just returns an HRESULT, we won't be storing the error info anywhere

Reply:
My thoughts too, I'll switch to that.

Reply:
I switched over and reworked the code so `getErrorDescription` doesn't modify the thread error state.